### PR TITLE
Option for ObstacleLayer to not overwrite StaticLayer's unknown parts

### PR DESCRIPTION
**Pull request by RBT22**
The documentation for [#3612](https://github.com/ros-planning/navigation2/pull/3612)

### CODE REVIEW

**Comment by RBT22** [comment]
> Need to add to migration guide on website that this new feature exists to folks can use it!

Where should I put the migration guide?

> Need to add on configuration guide for costmap 2d the new setting so folks in the future can know its there

What should be this one? Isn't it enough to put info about it in the layer params docs?

File: configuration/packages/costmap-plugins/obstacle.rst [hunk 1]
```diff
@@ -49,7 +49,18 @@ Obstacle Layer Parameters
   ====== =======
 
   Description
-    Enum for method to add data to master costmap, default to maximum.
+    Enum for method to add data to master costmap. Must be 0, 1 or 2, default to 1 (see below).
+
+  | 0 - Overwrite: Overwrite master costmap with every valid observation.
+
+  | 1 - Max: Sets the new value to the maximum of the master_grid's value and this layer's value. 
+  | If the master value is NO_INFORMATION, it is overwritten. This is the default.
```
Comment (SteveMacenski) [inline configuration/packages/costmap-plugins/obstacle.rst:57]:
```suggestion
  | This is the default.
```

File: configuration/packages/costmap-plugins/voxel.rst [hunk 2]
```diff
@@ -104,7 +104,17 @@ Voxel Layer Parameters
   ====== =======
 
   Description
-    Enum for method to add data to master costmap, default to maximum.
+    Enum for method to add data to master costmap. Must be 0, 1 or 2, default to 1 (see below).
+
+  | 0 - Overwrite: Overwrite master costmap with every valid observation.
+
+  | 1 - Max: Sets the new value to the maximum of the master_grid's value and this layer's value. 
+  | If the master value is NO_INFORMATION, it is overwritten. This is the default.
```
Comment (SteveMacenski) [inline configuration/packages/costmap-plugins/voxel.rst:112]:
```suggestion
  | This is the default.
```

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
We also need an entry in the migration guide announcing this change

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**